saleor/order/calculations.py

```diff
@@ -0,0 +1,291 @@
+
+        total = price_line["total"]
+        order_line.undiscounted_total_price = total.undiscounted_price
+        order_line.total_price = total.price_with_discounts
+
+        order_line.tax_rate = price_line["tax_rate"]
+
+
+def _apply_tax_data(
+    order: Order, lines: Iterable[OrderLine], tax_data: TaxData
+) -> None:
+    def _quantize_price(net: Decimal, gross: Decimal) -> TaxedMoney:
+        currency = order.currency
+        return quantize_price(
+            TaxedMoney(net=Money(net, currency), gross=Money(gross, currency)),
+            currency,
+        )
+
+    order.total = _quantize_price(
+        net=tax_data.total_net_amount, gross=tax_data.total_gross_amount
+    )
+    order.shipping_price = _quantize_price(
+        net=tax_data.shipping_price_net_amount,
+        gross=tax_data.shipping_price_gross_amount,
+    )
+    order.shipping_tax_rate = tax_data.shipping_tax_rate
+
+    tax_lines = {line.id: line for line in tax_data.lines}
+    zipped_order_and_tax_lines = ((line, tax_lines[line.id]) for line in lines)
+
+    for (order_line, tax_line) in zipped_order_and_tax_lines:
+        order_line.unit_price = _quantize_price(
+            net=tax_line.unit_net_amount, gross=tax_line.unit_gross_amount
+        )
+        order_line.undiscounted_unit_price = (
+            order_line.unit_price + order_line.unit_discount
+        )
+
+        order_line.total_price = _quantize_price(
+            net=tax_line.total_net_amount, gross=tax_line.total_gross_amount
+        )
+        order_line.undiscounted_total_price = (
+            order_line.undiscounted_unit_price * order_line.quantity
+            if order_line.unit_discount
+            else order_line.total_price
+        )
+
+        order_line.tax_rate = tax_line.tax_rate
+
+
+def fetch_order_prices_if_expired(
+    order: Order,
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
+    force_update: bool = False,
+) -> Tuple[Order, Optional[Iterable[OrderLine]]]:
+    """Fetch order prices with taxes.
+
+    First calculate and apply all order prices with taxes separately,
+    then apply tax data as well if we receive one.
+
+    Prices can be updated only if force_update == True,
+    or if order price expiration time was exceeded
+    (which is settings.ORDER_PRICES_TTL if the prices are not invalidated).
+    """
+    if order.status not in ORDER_EDITABLE_STATUS:
+        return order, lines
+
+    if not force_update and order.price_expiration_for_unconfirmed < timezone.now():
+        return order, lines
+
+    if lines is None:
+        lines = list(order.lines.prefetch_related("variant__product"))
+    else:
+        prefetch_related_objects(lines, "variant__product")
+
+    manager_tax_data = _get_tax_data_from_manager(manager, order, lines)
+
+    if manager_tax_data is None:
+        return order, lines
+
+    _apply_tax_data_from_manager(manager_tax_data, order, lines)
```

Comment:
What is a reason for splitting logic between `_get_tax_data_from_manager` and `_apply_tax_data_from_manager`? 
Can 't we have a single method that will iterate over lines, call old tax methods for each line and immiedietly assign it to Order object?  Then we could call the new tax webhook. At the end call save on order and lines.